core/src/test/scala/integration/kafka/api/TransactionsTest.scala

```diff
@@ -352,6 +351,7 @@ class TransactionsTest extends KafkaServerTestHarness {
       }
     } finally {
       consumer.close()
+      producer1.close()
```

Comment:
That's an excellent suggestion. I created this JIRA to keep track of this, and will get to after we fix the existing blockers: https://issues.apache.org/jira/browse/KAFKA-5282